pkg/registry/extensions/daemonset/strategy.go

```diff
@@ -53,6 +53,7 @@ func (daemonSetStrategy) PrepareForCreate(ctx genericapirequest.Context, obj run
 	daemonSet.Status = extensions.DaemonSetStatus{}
 
 	daemonSet.Generation = 1
+	daemonSet.TemplateGeneration = 1
```

Comment:
We allow users to set this value on creation time, so only set this to 1 if it's unset (i.e. any value that's <=0).